pkg/apis/events/v1alpha1/pubsub_lifecycle.go

```diff
@@ -37,24 +43,42 @@ func (ps *PubSubStatus) InitializeConditions() {
 	pubSubCondSet.Manage(ps).InitializeConditions()
 }
 
-// MarkPullSubscriptionNotReady sets the condition that the underlying PullSubscription
-// source is not ready and why.
-func (ps *PubSubStatus) MarkPullSubscriptionNotReady(reason, messageFormat string, messageA ...interface{}) {
+// MarkPullSubscriptionFailed sets the condition that the underlying PullSubscription
+// source is False and why.
+func (ps *PubSubStatus) MarkPullSubscriptionFailed(reason, messageFormat string, messageA ...interface{}) {
 	pubSubCondSet.Manage(ps).MarkFalse(duckv1alpha1.PullSubscriptionReady, reason, messageFormat, messageA...)
 }
 
+// MarkPullSubscriptionNotConfigured sets the condition that the underlying PullSubscription
+// source has not been reconciled and why.
+func (ps *PubSubStatus) MarkPullSubscriptionNotConfigured() {
+	pubSubCondSet.Manage(ps).MarkUnknown(duckv1alpha1.PullSubscriptionReady, "PullSubscriptionNotConfigured", "PullSubscription has not yet been reconciled")
+}
+
 // MarkPullSubscriptionReady sets the condition that the underlying PullSubscription is ready.
 func (ps *PubSubStatus) MarkPullSubscriptionReady() {
 	pubSubCondSet.Manage(ps).MarkTrue(duckv1alpha1.PullSubscriptionReady)
 }
 
+// MarkPullSubscriptionReady sets the condition that the underlying PullSubscription is Unknown and why.
+func (ps *PubSubStatus) MarkPullSubscriptionUnknown(reason, messageFormat string, messageA ...interface{}) {
+	pubSubCondSet.Manage(ps).MarkUnknown(duckv1alpha1.PullSubscriptionReady, reason, messageFormat, messageA...)
+}
+
 func (ps *PubSubStatus) PropagatePullSubscriptionStatus(ready *apis.Condition) {
+	if ready == nil {
+		ps.MarkPullSubscriptionNotConfigured()
+		return
+	}
+
 	switch {
-	case ready == nil:
-		ps.MarkPullSubscriptionNotReady("PullSubscriptionNotReady", "PullSubscription has no Ready type status")
-	case ready.IsTrue():
+	case ready.Status == corev1.ConditionUnknown:
+		ps.MarkPullSubscriptionUnknown(ready.Reason, ready.Message)
+	case ready.Status == corev1.ConditionTrue:
 		ps.MarkPullSubscriptionReady()
-	case ready.IsFalse():
-		ps.MarkPullSubscriptionNotReady(ready.Reason, ready.Message)
+	case ready.Status == corev1.ConditionFalse:
+		ps.MarkPullSubscriptionFailed(ready.Reason, ready.Message)
+	default:
+		ps.MarkPullSubscriptionUnknown("PullSubscriptionUnknown", "The status of PullSubscription is invalid: %v", ready.Status)
```

Comment:
can it go in here? I guess is safer to have this default?

Reply:
I'm confused with this comment. What do you mean by "it"?

Reply:
Are you asking whether it's safer to have this default? I think so. Otherwise, we will mark the PullSubscriptionReady inaccurately or fail to mark it.

Reply:
yeah... I haven't seen that default in the serving example you sent. So I'm not sure if that condition status can be something different than ConditionFalse, ConditionTrue, or ConditionUnknown... If not, then it will never go through that default... 
But I think you are right, I'd rather leave it in case the user messes up with things there.